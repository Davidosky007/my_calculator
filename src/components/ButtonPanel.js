/* eslint-disable linebreak-style */
/* eslint-disable react/prop-types */
/* eslint-disable linebreak-style */
import React from 'react';
import propTypes from 'prop-types';
import Button from './Button';

const buttonGroups = {
  1: ['AC', '+/-', '%', 'รท'],
  2: ['7', '8', '9', 'x'],
  3: ['4', '5', '6', '-'],
  4: ['1', '2', '3', '+'],
  5: ['0', '.', '='],
};

const ButtonPanel = ({ handleClick }) => (
  <div className="btns">
    <div className="btn_grp">
      {buttonGroups[1].map((btn) => (
        <Button
          name={btn}
          handleClick={handleClick}
          key={btn}
        />
      ))}
    </div>
    <div className="btn_grp">
      {buttonGroups[2].map((btn) => (
        <Button
          name={btn}
          handleClick={handleClick}
          key={btn}
        />
      ))}
    </div>
    <div className="btn_grp">
      {buttonGroups[3].map((btn) => (
        <Button
          name={btn}
          handleClick={handleClick}
          key={btn}
        />
      ))}
    </div>
    <div className="btn_grp">
      {buttonGroups[4].map((btn) => (
        <Button
          name={btn}
          handleClick={handleClick}
          key={btn}
        />
      ))}
    </div>
    <div className="btn_grp2">
      {buttonGroups[5].map((btn) => (
        <Button
          name={btn}
          handleClick={handleClick}
          key={btn}
        />
      ))}
    </div>

  </div>
);
ButtonPanel.protoTypes = {
  handleClick: propTypes.func,
};

ButtonPanel.defaultProps = {
  handleClick: null,
};

export default ButtonPanel;
