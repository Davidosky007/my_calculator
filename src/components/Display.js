/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledDisplay = styled.h2`
display: flex;
color: white;
font-weight: bold;
float: right;
`;

const StyledDisCompo = styled.section`
background-color: gray;
height: 100px;
padding: 15px;
width: 400px;
margin: 0 auto;
border-radius: 30px;
`;

const Display = (props) => {
  const { result } = props;
  return (
    <StyledDisCompo>
      <StyledDisplay>
        {result}
      </StyledDisplay>
    </StyledDisCompo>
  );
};

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
