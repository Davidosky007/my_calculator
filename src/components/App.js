/* eslint-disable linebreak-style */
import React, { useState } from 'react';
import styled from 'styled-components';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import Calculate from '../logic/calculate';

const StyledApp = styled.section`
display: inline-block;
width: 700px;
float: right;

`;

const App = () => {
  const [state, setState] = useState({
    total: '',
    next: '',
    operation: null,
  });

  const handleClick = (buttonName) => {
    const newState = Calculate(state, buttonName);
    setState(() => ({ ...newState }));
  };
  return (
    <>
      <h2 className="app_head">Let us do math!</h2>
      <StyledApp>
        <Display result={(state.next && state.next.toString()) || (state.total && state.total.toString()) || '0'} />
        {' '}
        <ButtonPanel handleClick={handleClick} />
      </StyledApp>
    </>
  );
};

export default App;
