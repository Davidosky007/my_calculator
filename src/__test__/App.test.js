/* eslint-disable linebreak-style */
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import App from '../components/App';
import Display from '../components/Display';
import ButtonPanel from '../components/ButtonPanel';

describe('App component test', () => {
  test('render App component correctly ', () => {
    const proof = create(<App />).toJSON();
    expect(proof).toMatchSnapshot();
  });
  test('displays heading', () => {
    render(<App />);
    const h2 = screen.getByText('Let us do math!');
    expect(h2).toBeInTheDocument();
  });
  test('render Display component correctly', () => {
    const display = create(<Display />).toJSON();
    expect(display).toMatchSnapshot();
  });
  test('render ButtonPannel component correctly', () => {
    const buttonPanel = create(<ButtonPanel />).toJSON();
    expect(buttonPanel).toMatchSnapshot();
  });
});
