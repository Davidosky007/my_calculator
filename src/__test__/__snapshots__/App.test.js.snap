// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test render App component correctly  1`] = `
Array [
  <h2
    className="app_head"
  >
    Let us do math!
  </h2>,
  <section
    className="sc-dlnjwi jPtfJn"
  >
    <section
      className="sc-gtsrHT ilhZDh"
    >
      <h2
        className="sc-bdnxRM gNjEKx"
      >
        0
      </h2>
    </section>
     
    <div
      className="btns"
    >
      <div
        className="btn_grp"
      >
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          AC
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          +/-
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          %
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          รท
           
        </button>
      </div>
      <div
        className="btn_grp"
      >
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          7
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          8
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          9
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          x
           
        </button>
      </div>
      <div
        className="btn_grp"
      >
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          4
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          5
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          6
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          -
           
        </button>
      </div>
      <div
        className="btn_grp"
      >
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          1
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          2
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          3
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          +
           
        </button>
      </div>
      <div
        className="btn_grp2"
      >
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          0
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          .
           
        </button>
        <button
          className="btn button"
          onClick={[Function]}
          type="button"
        >
          =
           
        </button>
      </div>
    </div>
  </section>,
]
`;

exports[`App component test render ButtonPannel component correctly 1`] = `
<div
  className="btns"
>
  <div
    className="btn_grp"
  >
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      AC
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      +/-
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      %
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      รท
       
    </button>
  </div>
  <div
    className="btn_grp"
  >
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      7
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      8
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      9
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      x
       
    </button>
  </div>
  <div
    className="btn_grp"
  >
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      4
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      5
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      6
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      -
       
    </button>
  </div>
  <div
    className="btn_grp"
  >
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      1
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      2
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      3
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      +
       
    </button>
  </div>
  <div
    className="btn_grp2"
  >
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      0
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      .
       
    </button>
    <button
      className="btn button"
      onClick={[Function]}
      type="button"
    >
      =
       
    </button>
  </div>
</div>
`;

exports[`App component test render Display component correctly 1`] = `
<section
  className="sc-gtsrHT ilhZDh"
>
  <h2
    className="sc-bdnxRM gNjEKx"
  >
    0
  </h2>
</section>
`;
